pkg/agent/cniserver/interface_configuration_linux.go

```diff
@@ -83,8 +83,58 @@ func moveIfToNetns(ifname string, netns ns.NetNS) error {
 	return nil
 }
 
-// configureContainerLinkSriov move VF to the container namesapce
-func (ic *ifConfigurator) configureContainerLinkSriov(
+func moveVFtoContainerNS(vfNetDevice string, containerID string, containerNetNS string, containerIfaceName string, mtu int, result *current.Result) error {
+        hostIface := result.Interfaces[0]
+        containerIface := result.Interfaces[1]
+
+        // Move VF to Container namespace
+        netns, err := ns.GetNS(containerNetNS)
+        if err != nil {
+               return fmt.Errorf("failed to open container netns %s: %+v", containerNetNS, err)
+        }
+        err = moveIfToNetns(vfNetDevice, netns)
+        if err != nil {
+               return fmt.Errorf("failed to move VF %+v to container netns %s: %+v", vfNetDevice, containerNetNS, err)
+        }
+        netns.Close()
+
+        if err := ns.WithNetNSPath(containerNetNS, func(hostNS ns.NetNS) error {
+                err = renameLink(vfNetDevice, containerIfaceName)
+                if err != nil {
+                       return fmt.Errorf("failed to rename VF netdevice as containerIfaceName %s: %+v", containerIfaceName, err)
+                }
+                link, err := netlink.LinkByName(containerIfaceName)
+                if err != nil {
+                        return fmt.Errorf("failed to find VF netdevice %s: %+v", containerIfaceName, err)
+                }
+                err = netlink.LinkSetMTU(link, mtu)
+                if err != nil {
+                        return fmt.Errorf("failed to set MTU for VF netdevice %s: %+v", containerIfaceName, err)
+                }
+                err = netlink.LinkSetUp(link)
+                if err != nil {
+                        return fmt.Errorf("failed to set link up to VF netdevice %s: %+v", containerIfaceName, err)
+                }
+                containerIface.Name = containerIfaceName
+                containerIface.Mac = link.Attrs().HardwareAddr.String()
+                containerIface.Sandbox = netns.Path()
+                klog.V(2).Infof("hostIface: %+v, containerIface: %+v", hostIface, containerIface)
+                klog.V(2).Infof("Configuring IP address for container %s", containerID)
+                // result.Interfaces must be set before this.
+                if err := ipam.ConfigureIface(containerIface.Name, result); err != nil {
+                        return fmt.Errorf("failed to configure IP address for container %s: %+v", containerID, err)
+                }
+                klog.V(2).Infof("ipam.ConfigureIface result: %+v, err: %+v", result, err)
+                return nil
+        }); err != nil {
+                return err
+        }
+
+        return nil
+}
+
+// configureContainerSriovLinkOnBridge moves VF to the container namesapce
```

Comment:
maybe it would help to improve this function comment so readers can understand the difference with `configureContainerSriovLink`.

Reply:
Done